Src/LiquidProjections.NEventStore/Subscription.cs

```diff
@@ -44,19 +45,109 @@ public void Start()
 
                 cancellationTokenSource = new CancellationTokenSource();
 #if DEBUG
-                LogProvider.GetCurrentClassLogger().Debug(() => $"Subscription {Id ?? "without ID"} has been started.");
+                LogProvider.GetCurrentClassLogger().Debug(() => $"Subscription {Id} has been started.");
 #endif
 
-                Task = Task.Factory
-                    .StartNew(
-                        RunAsync,
+                SubscriptionInfo info = new SubscriptionInfo
+                {
+                    Id = Id,
+                    Subscription = this
+                };
+
+                Task = Task.Factory.StartNew(async () =>
+                        {
+                            try
+                            {
+                                await RunAsync(info);
+                            }
+                            catch (OperationCanceledException)
+                            {
+                                // Do nothing.
+                            }
+                            catch (Exception exception)
+                            {
+                                LogProvider.GetCurrentClassLogger().FatalException(
+                                    "NEventStore polling task has failed. Event subscription has been cancelled.",
+                                    exception);
+                            }
+                        },
                         cancellationTokenSource.Token,
                         TaskCreationOptions.DenyChildAttach | TaskCreationOptions.LongRunning,
                         TaskScheduler.Default)
                     .Unwrap();
             }
         }
 
+        private async Task RunAsync(SubscriptionInfo info)
+        {
+            const int offsetToDetectAheadSubscriber = 1;
+            long actualRequestedLastCheckpoint = lastProcessedCheckpoint;
+            lastProcessedCheckpoint = lastProcessedCheckpoint > 0 ? lastProcessedCheckpoint - offsetToDetectAheadSubscriber : 0;
+            bool firstRequestAfterSubscribing = true;
+
+            while (!cancellationTokenSource.IsCancellationRequested)
```

Comment:
Maybe it makes sense to split this loop so that the ahead detection logic does not interfere with the usual operation.